app/src/main/java/com/itsaky/androidide/data/common/IDEContainer.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ *  This file is part of AndroidIDE.
+ *
+ *  AndroidIDE is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  AndroidIDE is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *   along with AndroidIDE.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.itsaky.androidide.data.common
+
+import android.content.Context
+import com.itsaky.androidide.data.projectInfo.OfflineProjectInfoRepository
+import com.itsaky.androidide.data.projectInfo.ProjectInfoRepository
+
+interface IDEContainer {
+
+  val projectInfoRepository: ProjectInfoRepository
+}
```

Comment:
Do we really need a separate class just to store the `ProjectInfoRepository`? Prefer registering this repository object with the `Lookup` API so that it could be retrieved anytime and without the help of the `Application` instance.